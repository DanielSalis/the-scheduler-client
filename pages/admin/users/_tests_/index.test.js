import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'
import AdminUserPage from '../index.vue'
import UsersList from '~/components/users/users-list.vue'
import UsersAdd from '~/components/users/users-add.vue'

// Mock the layout
const MockLayout = {
  template: '<div><slot /></div>',
}

describe('AdminUserPage.vue', () => {
  let localVue
  let vuetify

  beforeEach(() => {
    localVue = createLocalVue()
    vuetify = new Vuetify()
    localVue.use(Vuetify)
  })

  it('renders the component and matches snapshot', () => {
    const wrapper = shallowMount(AdminUserPage, {
      localVue,
      vuetify,
      mocks: {
        $axios: {
          get: jest.fn(() => Promise.resolve({ data: [] })),
        },
      },
      stubs: {
        layout: MockLayout,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders UsersList and UsersAdd components based on data', async () => {
    const wrapper = shallowMount(AdminUserPage, {
      localVue,
      vuetify,
      mocks: {
        $axios: {
          get: jest.fn(() => Promise.resolve({
            data: [
              { id: 1, name: 'John Doe' },
              { id: 2, name: 'Jane Smith' }
            ]
          })),
        },
      },
      stubs: {
        layout: MockLayout,
      },
    })

  })

  it('computes headers correctly', async () => {
    const wrapper = shallowMount(AdminUserPage, {
      localVue,
      vuetify,
      data() {
        return {
          usersData: [
            { id: 1, name: 'John Doe' },
            { id: 2, name: 'Jane Smith' }
          ]
        }
      },
      mocks: {
        $axios: {
          get: jest.fn(() => Promise.resolve({
            data: [
              { id: 1, name: 'John Doe' },
              { id: 2, name: 'Jane Smith' }
            ]
          })),
        },
      },
      stubs: {
        layout: MockLayout,
      },
    })

    const expectedHeaders = [
      { text: 'ID', value: 'id' },
      { text: 'NAME', value: 'name' },
      { text: 'ACTIONS', value: 'actions', sortable: false },
    ]

    expect(wrapper.vm.headers).toEqual(expectedHeaders)
  })
})
