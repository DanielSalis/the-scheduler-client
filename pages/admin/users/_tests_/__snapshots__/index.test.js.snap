// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminUserPage.vue renders the component and matches snapshot 1`] = `
"<gcontainer-stub class=\\"admin-user-page-container\\">
  <v-card-stub loaderheight=\\"4\\" tag=\\"div\\">
    <v-toolbar-stub color=\\"white\\" tag=\\"header\\" extensionheight=\\"48\\" flat=\\"true\\" src=\\"\\">
      <v-toolbar-title-stub tag=\\"div\\">Usu√°rios</v-toolbar-title-stub>
    </v-toolbar-stub>
    <v-tabs-stub activeclass=\\"\\" nexticon=\\"$next\\" previcon=\\"$prev\\" slidersize=\\"2\\">
      <v-tab-stub ripple=\\"true\\">
        <v-icon-stub left=\\"\\">
          mdi-table
        </v-icon-stub>
        Listar
      </v-tab-stub>
      <v-tab-stub ripple=\\"true\\">
        <v-icon-stub left=\\"\\">
          mdi-plus
        </v-icon-stub>
        Adicionar
      </v-tab-stub>
      <!---->
      <v-tab-item-stub>
        <usersadd-stub></usersadd-stub>
      </v-tab-item-stub>
      <v-tab-item-stub>
        <v-card-stub loaderheight=\\"4\\" tag=\\"div\\" flat=\\"true\\">
          <v-card-text-stub tag=\\"div\\">
            <p>
              Fusce a quam. Phasellus nec sem in justo pellentesque facilisis. Nam eget dui. Proin viverra, ligula sit amet ultrices semper, ligula arcu tristique sapien, a accumsan nisi mauris ac eros. In dui magna, posuere eget, vestibulum et, tempor auctor, justo.
            </p>
            <p class=\\"mb-0\\">
              Cras sagittis. Phasellus nec sem in justo pellentesque facilisis. Proin sapien ipsum, porta a, auctor quis, euismod ut, mi. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nam at tortor in tellus interdum sagittis.
            </p>
          </v-card-text-stub>
        </v-card-stub>
      </v-tab-item-stub>
    </v-tabs-stub>
  </v-card-stub>
</gcontainer-stub>"
`;
